src/luascript.cpp

```diff
@@ -13777,66 +13812,927 @@ int LuaScriptInterface::luaPartySetSharedExperience(lua_State* L)
 // Spells
 int LuaScriptInterface::luaSpellCreate(lua_State* L)
 {
-	// Spell(words, name or id)
-	InstantSpell* spell = nullptr;
+	// Spell(words, name or id) to get an existing spell
+	// Spell(type) ex: Spell(SPELL_INSTANT) or Spell(SPELL_RUNE) to create a new spell
+	SpellType_t type = getNumber<SpellType_t>(L, 2);
+	if (isString(L, 2)) {
+		std::string tmp = asLowerCaseString(getString(L, 2));
+		if (tmp == "instant") {
+			type = SPELL_INSTANT;
+		} else if (tmp == "rune") {
+			type = SPELL_RUNE;
+		}
+	}
+
+	if (type) {
```

Comment:
This check seems a little weird, considering the fact we have a `switch` below, IMO you should just add a case to the switch that handles that value explicitly. This gets rid of one level of nesting, thus improving readability.

Reply:
Well as this function consists of either creating spells or returning spell userdata, I'm just checking here beforehand if it's a creation type and then use the switch to work accordingly, if it's not a type of creating it should skip the switch anyway.

Reply:
Sure it's equivalent, but using just the `switch` is easier to read, because it has one level of indentation less.

Reply:
I cannot possibly remove it because when parameter is a string then type = 0 which results that the switch will throw it as `SPELL_UNDEFINED` which we don't want.

Reply:
```	switch (type) {
		case SPELL_INSTANT: {
			InstantSpell* spell = new InstantSpell(getScriptEnv()->getScriptInterface());
			if (spell) {
				spell->fromLua = true;
				pushUserdata<InstantSpell>(L, spell);
				setMetatable(L, -1, "Spell");
				spell->spellType = SPELL_INSTANT;
			} else {
				lua_pushnil(L);
			}
			return 1;
		}
		case SPELL_RUNE: {
			RuneSpell* spell = new RuneSpell(getScriptEnv()->getScriptInterface());
			if (spell) {
				spell->fromLua = true;
				pushUserdata<RuneSpell>(L, spell);
				setMetatable(L, -1, "Spell");
				spell->spellType = SPELL_RUNE;
			} else {
				lua_pushnil(L);
			}
			return 1;
		}
		case SPELL_UNDEFINED: {
			break;
		}
		default: {
			lua_pushnil(L);
			return 1;
		}
	}
```

Reply:
that doesn't work either, if I push now like `Spell(220)` it'll go to default and give me nil, which should actually give me the userdata of spell with id 220
I don't see one check with more indentation worse if it does exactly the job it should.

Reply:
I think you're confused what `if (type) {` does...
It checks whether the enum is greater than 0. So `Spell(220)` would cause that condition to evaluate to `true`. Then it goes into the `switch` and jumps into the `default` case. Both snippets are equivalent.

Reply:
I'll just remove the switch it's just `SPELL_INSTANT` and `SPELL_RUNE` anyway and rather check at the beginning if `lua_gettop(L)` returns 1 and then `nil` it with an error message